src/cleanup.ts

```diff
@@ -7,30 +7,36 @@ import * as logging from "./logging";
 
 export const FILE_MANIFEST = `${PROJEN_DIR}/files.json`;
 
-export function cleanup(dir: string, exclude: string[]) {
+export function cleanup(dir: string, newFiles: string[], exclude: string[]) {
   try {
-    for (const f of findGeneratedFiles(dir, exclude)) {
-      fs.removeSync(f);
+    const manifestFiles = getFilesFromManifest(dir);
+    if (manifestFiles.length > 0) {
+      // Use `FILE_MANIFEST` to remove files that are no longer managed by projen
+      cleanOrphanedFiles(dir, manifestFiles, newFiles);
+    } else {
+      // Remove all files managed by projen with legacy logic
+      removeFiles(findGeneratedFiles(dir, exclude));
     }
```

Comment:
Can we normalize these branches like so:
```suggestion
    if (manifestFiles.length > 0) {
      // Use `FILE_MANIFEST` to remove files that are no longer managed by projen
      removeFiles(findOrphanedFiles(dir, manifestFiles, newFiles));
    } else {
      // Remove all files managed by projen with legacy logic
      removeFiles(findGeneratedFiles(dir, exclude));
    }
```